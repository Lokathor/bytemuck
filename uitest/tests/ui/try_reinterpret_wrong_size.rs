use bytemuck::TryReinterpret;

fn main() {
  let _: u16 = 0u8.try_reinterpret().unwrap();
  let _: u32 = 0u16.try_reinterpret().unwrap();
  let _: u32 = 0u64.try_reinterpret().unwrap();
}
