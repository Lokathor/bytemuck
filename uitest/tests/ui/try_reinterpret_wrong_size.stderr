error[E0080]: evaluation of `<u8 as bytemuck::TryReinterpret<u16>>::ASSERT` failed
 --> $WORKSPACE/src/cast.rs
  |
  | /     assert!(
  | |       size_of::<Src>() == size_of::<Dst>(),
  | |       "attempted conversion between types of different sizes"
  | |     );
  | |_____^ the evaluated program panicked at 'attempted conversion between types of different sizes', $WORKSPACE/src/cast.rs:102:5
  |
  = note: this error originates in the macro `$crate::panic::panic_2015` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

note: the above error was encountered while instantiating `fn <u8 as bytemuck::TryReinterpret<u16>>::try_reinterpret`
 --> tests/ui/try_reinterpret_wrong_size.rs:4:16
  |
4 |   let _: u16 = 0u8.try_reinterpret().unwrap();
  |                ^^^^^^^^^^^^^^^^^^^^^

error[E0080]: evaluation of `<u16 as bytemuck::TryReinterpret<u32>>::ASSERT` failed
 --> $WORKSPACE/src/cast.rs
  |
  | /     assert!(
  | |       size_of::<Src>() == size_of::<Dst>(),
  | |       "attempted conversion between types of different sizes"
  | |     );
  | |_____^ the evaluated program panicked at 'attempted conversion between types of different sizes', $WORKSPACE/src/cast.rs:102:5
  |
  = note: this error originates in the macro `$crate::panic::panic_2015` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

note: the above error was encountered while instantiating `fn <u16 as bytemuck::TryReinterpret<u32>>::try_reinterpret`
 --> tests/ui/try_reinterpret_wrong_size.rs:5:16
  |
5 |   let _: u32 = 0u16.try_reinterpret().unwrap();
  |                ^^^^^^^^^^^^^^^^^^^^^^

error[E0080]: evaluation of `<u64 as bytemuck::TryReinterpret<u32>>::ASSERT` failed
 --> $WORKSPACE/src/cast.rs
  |
  | /     assert!(
  | |       size_of::<Src>() == size_of::<Dst>(),
  | |       "attempted conversion between types of different sizes"
  | |     );
  | |_____^ the evaluated program panicked at 'attempted conversion between types of different sizes', $WORKSPACE/src/cast.rs:102:5
  |
  = note: this error originates in the macro `$crate::panic::panic_2015` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

note: the above error was encountered while instantiating `fn <u64 as bytemuck::TryReinterpret<u32>>::try_reinterpret`
 --> tests/ui/try_reinterpret_wrong_size.rs:6:16
  |
6 |   let _: u32 = 0u64.try_reinterpret().unwrap();
  |                ^^^^^^^^^^^^^^^^^^^^^^
